src/k8s/api/v1/types.go

```diff
@@ -24,6 +24,19 @@ type BootstrapConfig struct {
 	DatastoreClientCert string   `yaml:"datastore-client-crt,omitempty"`
 	DatastoreClientKey  string   `yaml:"datastore-client-key,omitempty"`
 	ExtraSANs           []string `yaml:"extrasans,omitempty"`
+
+	CACert                     string `yaml:"ca-crt,omitempty"`
+	CAKey                      string `yaml:"ca-key,omitempty"`
+	FrontProxyCACert           string `yaml:"front-proxy-ca-crt"`
+	FrontProxyCAKey            string `yaml:"front-proxy-ca-key"`
+	APIServerKubeletClientCert string `yaml:"apiserver-kubelet-client-crt"`
+	APIServerKubeletClientKey  string `yaml:"apiserver-kubelet-client-key"`
+	ServiceAccountKey          string `yaml:"service-account-key"`
+
+	APIServerCert string `yaml:"apiserver-crt,omitempty"`
+	APIServerKey  string `yaml:"apiserver-key,omitempty"`
+	KubeletCert   string `yaml:"kubelet-crt,omitempty"`
+	KubeletKey    string `yaml:"kubelet-key,omitempty"`
```

Comment:
so here we go --- this config blob continues to grow unchecked.  

it looks to me that you've got "Keys" and "Certs".  Maybe you can make an object called `CertKey`  struct?  

Then include numerous of those here. 

```go
CA:               CertKey
FrontProxy:       CertKey
ServiceAccount:   CertKey
APIServer:        CertKey
APIServerKubelet: CertKey
Kubelet:          CertKey
```

Another question?  what's the different between `APIServerKubelet` and `Kubelet`?  Is the latter for the workers?

Reply:
`APIServerKubelet` is the client certificate that kube-apiserver uses to authenticate requests to the `kubelet`, e.g. when doing things like `kubectl logs`. `APIServer` is the server certificate that serves TLS on kube-apiserver

Having a `CertKey` is also useful, but then there are things like `ServiceAccountKey` which is just a private key. To avoid dealing with this, we have been using flat structs instead. Why flat? No implied structure is easier to iterate with than some pre-existing (and possibly misleading) notion of structure.